src/Elasticsearch.Net/Connection/HttpConnection.cs

```diff
@@ -194,42 +194,15 @@ private static void TimeoutCallback(object state, bool timedOut)
 			var builder = new ResponseBuilder<TReturn>(requestData, cancellationToken);
 			try
 			{
-				var request = this.CreateHttpWebRequest(requestData);
-				cancellationToken.Register(()=>request.Abort());
 				var data = requestData.PostData;
-
-				if (data != null)
+				var request = this.CreateHttpWebRequest(requestData);
+				using (cancellationToken.Register(() => request.Abort()))

```

Comment:
Does this also need to include `await GetResponseAsync(requestData, request, builder);` in the using block?